astropy/timeseries/downsample.py

```diff
@@ -10,7 +10,50 @@
 from astropy.timeseries.sampled import TimeSeries
 from astropy.utils.exceptions import AstropyUserWarning
 
-__all__ = ["aggregate_downsample"]
+__all__ = ["aggregate_downsample", "nanmean"]
+
+
+def nanmean_reduceat(data, indices):
+    mask = np.isnan(data)
+
+    if mask.any():  # If there are NaNs
+        # Create a writeable copy and mask NaNs
+        data_copy = data.copy()
+        data_copy[mask] = 0
+        count_data = np.add.reduceat(~mask, indices)
+        count_data = count_data.astype(float)
+        count_data[count_data == 0] = np.nan
+    else:
+        data_copy = data
+        # Derive counts from indices
+        count_data = np.diff(indices, append=len(data))
+        count_data[count_data <= 0] = 1
+
+    sum_data = np.add.reduceat(data_copy, indices)
+    nanmean = sum_data / count_data
+    return nanmean
+
+
+def nanmean(*args, **kwargs):
+    """
+    This function is a wrapper around `numpy.nanmean` and behaves identically to it.
+    Additionally, it includes the `reduceat` method, which performs grouped mean calculations
+    over slices of the data, defined by a set of indices, while ignoring NaN values.
+    The `reduceat` method behaves identical to `numpy.ufunc.reduceat` except that
+    parameters ``axis``, ``dtype``, and ``out`` are not supported.
+    For further functionality take a look at the ``See also`` section.
+
+    See Also
+    --------
+    numpy.nanmean
+    numpy.ufunc.reduceat
+    """
+    result = np.nanmean(*args, **kwargs)
+    return result
```

Comment:
No point binding a variable just to immediately return it, IMO
```suggestion
    return np.nanmean(*args, **kwargs)
```